src/UglyToad.PdfPig.DocumentLayoutAnalysis/ReadingOrderDetector/RenderingReadingOrderDetector.cs

```diff
@@ -17,14 +18,28 @@ public class RenderingReadingOrderDetector : IReadingOrderDetector
         /// <summary>
         /// Gets the blocks in reading order and sets the <see cref="TextBlock.ReadingOrder"/>.
         /// </summary>
-        /// <param name="textBlocks">The <see cref="TextBlock"/>s to order.</param>
-        public IEnumerable<TextBlock> Get(IReadOnlyList<TextBlock> textBlocks)
+        /// <param name="blocks">The <see cref="TextBlock"/>s to order.</param>
+        /// <returns>the orignal list if type is not <see cref="TextBlock"/></returns>
+        public IEnumerable<TBlock> Get<TBlock>(IEnumerable<TBlock> blocks)
+             where TBlock : IBoundingBox
+        {
+            if (typeof(TBlock) != typeof(TextBlock))
```

Comment:
Should we throw here instead?

This reading order detector would only work with type TextBlock. Should we throw when the type is not that?